integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ItMiiSampleHelper.java

```diff
@@ -0,0 +1,305 @@
+      execTestScriptAndAssertSuccess(
+          DomainType.JRF,
+          "-initial-image,-check-image-and-push,-initial-main",
+          "Initial use case failed"
+      );
+    }
+  }
+
+  /**
+   * Test MII sample WLS or JRF update1 use case.
+   */
+  public void callUpdate1UseCase() {
+    String useAuxiliaryImage = (getImageType().equalsIgnoreCase(ImageType.AUX.toString()))  ? "true" : "false";
+    envMap.put("DO_AI", useAuxiliaryImage);
+
+    if (getDomainType().equalsIgnoreCase(DomainType.WLS.toString())) {
+      execTestScriptAndAssertSuccess("-update1", "Update1 use case failed");
+    } else if (getDomainType().equalsIgnoreCase(DomainType.JRF.toString())) {
+      execTestScriptAndAssertSuccess(DomainType.JRF,"-update1", "Update1 use case failed");
+    }
+  }
+
+  /**
+   * Test MII sample WLS or JRF update2 use case.
+   */
+  public void callUpdate2UseCase() {
+    String useAuxiliaryImage = (getImageType().equalsIgnoreCase(ImageType.AUX.toString()))  ? "true" : "false";
+    envMap.put("DO_AI", useAuxiliaryImage);
+
+    if (getDomainType().equalsIgnoreCase(DomainType.WLS.toString())) {
+      execTestScriptAndAssertSuccess("-update2", "Update2 use case failed");
+    } else if (getDomainType().equalsIgnoreCase(DomainType.JRF.toString())) {
+      execTestScriptAndAssertSuccess(DomainType.JRF,"-update2", "Update2 use case failed");
+    }
+  }
+
+  /**
+   * Test MII sample WLS or JRF update3 use case.
+   */
+  public void callUpdate3UseCase() {
+    String useAuxiliaryImage = (getImageType().equalsIgnoreCase(ImageType.AUX.toString()))  ? "true" : "false";
+    String imageName = (getDomainType().equalsIgnoreCase(DomainType.WLS.toString()))
+        ? MII_SAMPLE_WLS_IMAGE_NAME_V2 : MII_SAMPLE_JRF_IMAGE_NAME_V2;
+    envMap.put("MODEL_IMAGE_NAME", imageName);
+    envMap.put("DO_AI", useAuxiliaryImage);
+
+    if (getDomainType().equalsIgnoreCase(DomainType.WLS.toString())) {
+      execTestScriptAndAssertSuccess("-update3-image,-check-image-and-push,-update3-main", "Update3 use case failed");
+    } else if (getDomainType().equalsIgnoreCase(DomainType.JRF.toString())) {
+      execTestScriptAndAssertSuccess(
+          DomainType.JRF,
+          "-update3-image,-check-image-and-push,-update3-main",
+          "Update3 use case failed"
+      );
+    }
+  }
+
+  /**
+   * Test MII sample WLS or JRF update4 use case.
+   */
+  public void callUpdate4UseCase() {
+    String useAuxiliaryImage = (getImageType().equalsIgnoreCase(ImageType.AUX.toString()))  ? "true" : "false";
+    envMap.put("DO_AI", useAuxiliaryImage);
+
+    if (getDomainType().equalsIgnoreCase(DomainType.WLS.toString())) {
+      execTestScriptAndAssertSuccess("-update4", "Update4 use case failed");
+    } else if (getDomainType().equalsIgnoreCase(DomainType.JRF.toString())) {
+      execTestScriptAndAssertSuccess(DomainType.JRF,"-update4", "Update4 use case failed");
+    }
```

Comment:
This code would be more compact if 'getDomainType' returned the DomainType enum. 

Then 

```
    if (getDomainType().equalsIgnoreCase(DomainType.WLS.toString())) {
      execTestScriptAndAssertSuccess("-update4", "Update4 use case failed");
    } else if (getDomainType().equalsIgnoreCase(DomainType.JRF.toString())) {
      execTestScriptAndAssertSuccess(DomainType.JRF,"-update4", "Update4 use case failed");
    }
```

would simply become:

```
    execTestScriptAndAssertSuccess(getDomainType(), "-update4", "Update4 use case failed");
```

Ditto for the similar code in the other methods...

Reply:
changed